evervault-enclaves/src/main/java/com/evervault/sdk/enclaves/AttestationDocCache.kt

```diff
@@ -1,76 +1,103 @@
-import android.content.ContentValues.TAG
-import java.io.IOException
-import java.util.concurrent.locks.ReentrantReadWriteLock
-import kotlin.concurrent.read
-import kotlin.concurrent.write
+import android.util.Base64
+import android.util.Log
 import kotlinx.coroutines.*
 import kotlinx.serialization.SerialName
-import okhttp3.OkHttpClient
-import okhttp3.Request
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.json.Json
-import android.util.Base64
-import android.util.Log
-import java.lang.Exception
+import okhttp3.Interceptor.*
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.write
+
 
 @Serializable
 data class AttestationDoc(
     @SerialName("attestation_doc") val attestationDoc: String,
 )
 @OptIn(DelicateCoroutinesApi::class)
-class AttestationDocCache(private val enclaveName: String, private val appUuid: String) {
+class AttestationDocCache(
+    private val enclaveName: String,
+    private val appUuid: String,
+    private val enclaveHostname: String? = "enclave.evervault.com",
+) {
     private var attestationDoc: ByteArray = ByteArray(0)
     private val lock = ReentrantReadWriteLock()
+    private val maxAttempts = 3
+    private val initialDelayMs = 1000L
+    private var pollingJob: Job? = null
+    private val client: OkHttpClient = OkHttpClient.Builder()
+        .callTimeout(10000, TimeUnit.MILLISECONDS)
+        .connectTimeout(10000, TimeUnit.MILLISECONDS)
+        .readTimeout(10000, TimeUnit.MILLISECONDS)
+        .build()
 
```

Comment:
should the timeouts be configurable?

Reply:
I don't believe so, these are the okhttp clients a relic or me testing. Removed to revert to the defaults.